science/nds2-client/Portfile

```diff
@@ -13,117 +16,169 @@ long_description \
+depends_lib-append port:sqlite3
+
+default_variants  +gssapi
 
-default_variants  +doc +gssapi +swig_python27 +swig_java
+cmake.out_of_source yes
 
-configure.javac   /usr/bin/javac -source 1.5 -target 1.5
-configure.args    --disable-silent-rules \
-                  --disable-doc \
-                  --disable-swig-python \
-                  --disable-swig-octave \
-                  --disable-swig-java \
-                  --without-sasl \
-                  --without-gssapi
+configure.args    -DPYTHON=false \
+                  -DPYTHON_EXECUTABLE=false \
+                  -DENABLE_SWIG_JAVA=no \
+                  -DENABLE_SWIG_MATLAB=no \
+                  -DENABLE_SWIG_OCTAVE=no \
+                  -DENABLE_SWIG_PYTHON=no \
+                  -DWITH_SASL=no \
+                  -DWITH_GSSAPI=no
 
-set pythons_suffixes {27 34}
+variant sasl description "Use cyrus-sasl2 for authentication" conflicts gssapi {
 
-set pythons_ports {}
-foreach s ${pythons_suffixes} {
-    lappend pythons_ports swig_python${s}
+    configure.args-replace          -DWITH_SASL=no -DWITH_SASL=${prefix}
+    depends_lib-append              port:cyrus-sasl2
 }
 
-variant doc description "Enable HTML documentation" {
+variant gssapi description "Use kerberos5 gssapi for authentication" conflicts sasl {
 
-    depends_build-append            port:docbook-xsl \
-                                    port:libxslt
-    configure.args-strsed           s/--disable-doc/--enable-doc/
+    configure.args-replace          -DWITH_GSSAPI=no -DWITH_GSSAPI=${prefix}
+    depends_lib-append              port:kerberos5
 
 }
 
-foreach s ${pythons_suffixes} {
-    set p python${s}
-    set v [string index ${s} 0].[string index ${s} 1]
-    set i [lsearch -exact ${pythons_ports} swig_${p}]
-    set c [lreplace ${pythons_ports} ${i} ${i}]
-    set d ${frameworks_dir}/Python.framework/Versions/${v}/lib/python${v}
-    eval [subst {
-        variant swig_${p} description "Enable SWIG Python interface for Python ${v}" conflicts ${c} {
+test.run           yes
+test.cmd           ${prefix}/bin/ctest
+test.target        -R '.*'
 
-            depends_build-append    port:swig-python
-            depends_lib-append      port:${p} port:py${s}-numpy
-            configure.args-strsed   s/--disable-swig-python/--enable-swig-python/
-            destroot.args-append    pythondir="${d}" pyexecdir="${d}"
+use_parallel_build yes
 
-        }
-    }]
-}
+livecheck.type    regex
+livecheck.url     ${master_sites}
+livecheck.regex   {nds2-client-(\d+(?:\.\d+)*).tar.gz}
 
-variant swig_octave description "Enable SWIG Octave interface" {
+#========================================================================
+# Create subports for Java
+#========================================================================
 
-    depends_build-append            port:swig-octave
-    configure.args-strsed           s/--disable-swig-octave/--enable-swig-octave/
+subport ${name}-java {
+  categories-prepend    java
```

Comment:
The first category of a port should be the the name of the folder in which the Portfile is located, so this should probably be categories-append. Try `port lint --nitpick nds2-client-java`, it might warn about this.